### PR TITLE
Refresh of Java SDK index page

### PR DESCRIPTION
**Pull request by langsmith**
This pr addresses a long overdue refresh and re-ordering of https://docs.mapbox.com/android/java/overview. 

I've removed the pre `3.0.0` packages and text because I can't imagine anyone's still using those versions. Having pre/post stuff on this page could lead to unneeded confusion about which to go with, the differences, etc.

screencapture-localhost-8080-android-java-overview-2019-07-09-15_05_57

### CODE REVIEW

File: src/pages/java/overview/index.md [hunk 1]
```diff
@@ -113,38 +61,66 @@ If your project's using Maven instead of Gradle, you can add the dependency insi
...
 ```
 
-> **Note:** ProGuard directives are included in the Android dependencies to preserve the required classes.
+### Access tokens
+
+You'll need to have a Mapbox access token if you plan to use any of the `mapbox-sdk-services` wrappers around Mapbox APIs. An access token isn't needed if you plan to use the Mapbox Java SDK for GeoJSON parsing or Turf calculations. Learn more about [how access tokens work](https://www.mapbox.com/help/create-api-access-token/).
+
+### Point object
+
+Throughout the APIs exposed within the Java SDK, you'll notice the common usage of the `Point` object to represent a coordinate. Longitude followed by latitude is the order of a `Point`'s coordinate pair. If you use the Maps SDK with the Java SDK, you'll notice that the `LatLng` object has the reverse order; latitude comes before the longitude value. Creating a new `Point` is fairly straightforward and can be converted from a `LatLng` object if you are using the Mapbox Maps SDK for Android.
```
Comment (colleenmcginnis) [inline src/pages/java/overview/index.md]:
```suggestion
The Java SDK uses the `Point` object to represent a geographical coordinate.  `Point` objects contain two values, longitude and latitude, in that order. This order is the reverse of the `LatLng` object in the Mapbox Maps SDK for Android. If you are using the Maps SDK and the Java SDK together, you will need to convert  between `Point` and `LatLng` objects.
```

File: src/pages/java/overview/index.md [hunk 2]
```diff
@@ -113,38 +61,66 @@ If your project's using Maven instead of Gradle, you can add the dependency insi
...
+
+The Java SDK will try to detect if your coordinates are outside the accepted range. This range is between -180 to 180 for longitudes and -90 to 90 for latitude. If the coordinates are outside the accepted range, the Java SDK will log a warning.
```
Comment (colleenmcginnis) [inline src/pages/java/overview/index.md]: I don't think we need this paragraph.

File: src/pages/java/overview/index.md [hunk 3]
```diff
@@ -113,38 +61,66 @@ If your project's using Maven instead of Gradle, you can add the dependency insi
...
+
+kotlin={`
+val singlePoint = Point.fromLngLat(LONGITUDE, LATITUDE);
+`}
+/>
+}}
+
+### Support and contributions
+
+- If you are looking for support using this SDK, either reach out through [Stack Overflow](https://stackoverflow.com/questions/tagged/mapbox+android) or through our [contact page](https://www.mapbox.com/contact/).
```
Comment (colleenmcginnis) [inline src/pages/java/overview/index.md]: I think this section should be simplified to something like: 

>## Support and contributions
>- For support using the Java SDK, browse [Stack Overflow](https://stackoverflow.com/questions/tagged/mapbox+android) or [contact support](https://support.mapbox.com).
>- For reproducible bugs and feature requests, open an issue in [the Java SDK GitHub repository](https://github.com/mapbox/mapbox-java).
>- To contribute to this project, see the [contributing guidelines on GitHub](https://github.com/mapbox/mapbox-java/blob/master/CONTRIBUTING.md).

File: src/pages/java/overview/index.md [hunk 4]
```diff
@@ -113,38 +61,66 @@ If your project's using Maven instead of Gradle, you can add the dependency insi
...
-implementation 'com.mapbox.mapboxsdk:mapbox-android-services:VERSION_NUMBER'
-implementation 'com.mapbox.mapboxsdk:mapbox-android-telemetry:VERSION_NUMBER'
-implementation 'com.mapbox.mapboxsdk:mapbox-android-ui:VERSION_NUMBER'
+implementation 'com.mapbox.mapboxsdk:mapbox-sdk-core:{{constants.JAVA_SDK_VERSION }}'
 ```
 
-> **Note:** ProGuard directives are included in the Android dependencies to preserve the required classes.
+### Access tokens
+
+You'll need to have a Mapbox access token if you plan to use any of the `mapbox-sdk-services` wrappers around Mapbox APIs. An access token isn't needed if you plan to use the Mapbox Java SDK for GeoJSON parsing or Turf calculations. Learn more about [how access tokens work](https://www.mapbox.com/help/create-api-access-token/).
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/index.md]: Changed to 

An access token **is not** needed if you plan to use the Mapbox Java SDK's GeoJSON library, Turf library, or certain parts of its Core library. Learn more about [how access tokens work](https://docs.mapbox.com/help/how-mapbox-works/access-tokens/).

File: src/pages/java/overview/index.md (hunk 2 above)
Comment (langsmith) [reply src/pages/java/overview/index.md]: Removed

File: src/pages/java/overview/index.md (hunk 3 above)
Comment (langsmith) [reply src/pages/java/overview/index.md]: Fixed

File: src/pages/java/overview/index.md [hunk 5]
```diff
@@ -103,7 +51,7 @@ To start developing your application using the Mapbox Java SDK, you'll need to f
 > **Note:** If your application is close or exceeds the 65k method count limit, you can mitigate this problem by specifying only the specific Mapbox Android Service APIs. See the selectively compiling APIs section below.
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/index.md]: Switched to the component

**Comment by langsmith** [comment]
Ok @colleenmcginnis , everything's been addressed 👁👁

**Pull request merged**